enhancements/microshift/etcd-as-a-transient-systemd-unit.md

```diff
@@ -0,0 +1,220 @@
+
+### Drawbacks
+
+We will have a more complicated build and an extra package to manage. We consider those acceptable tradeoffs to be able to deliver etcd fixes quickly and to decouple the dependencies that are causing MicroShift build issues.
+
+## Design Details
+
+### Open Questions
+
+#### RPM Build Cadence
+How do we get to pre-release version built out for every sprint?
+* How do we get fast-fix changes to customers that need/want them before a proper release?
+
+#### Naming
+Do we want to have it be just `etcd` or `microshift-etcd` or other?
+* Do we want to somehow indicate that this is specifically for MicroShift and not for general use; does that matter?
+
+#### Versioning
+
+* How should we version the etcd RPM; should we follow the OpenShift versioning for etcd.
+  * New versioning scheme that maps to an OpenShift/Upstream version
+  * Go with the etcd version: v3.5.4
+    * How do we reflect a patch or cherrypick update from OpenShift/etcd?
+  * OpenShift version: v4.12
+    * What is the release cadence for the etcd RPM/MicroShift as a whole? v4.12.1, v4.12.2, etc.
+    * How do we reflect an update to the etcd version for a particular OpenShift release branch e.g 4.12 gets updated from etcd 3.5.4 to 3.5.5?
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+In addition to standard end-to-end tests, we should also test for the main MicroShift binary dying/being killed and ensuring that etcd also comes down. This could be as simple as starting MicroShift via systemd, getting its pid, killing that pid, checking `systemctl status microshift-etcd` and ensuring it comes down.
+
+Another test could be if etcd has an ungraceful shutdown; do the same steps as above, but using microshift-etcd's pid. Etcd should automatically come back up via the same systemd unit.
+
+### Graduation Criteria
+
+TODO
+
+#### Dev Preview -> Tech Preview
+
+TODO
+
+#### Tech Preview -> GA
+
+TODO
+
+#### Removing a deprecated feature
+
+TODO
+
+### Upgrade / Downgrade Strategy
+
+TODO
+
+### Version Skew Strategy
+
+Currently, there is no need for a version skew strategy as etcd will be built and delivered as a part of the MicroShift RPM, so it will not be possible for the MicroShift and etcd versions to get out of sync.
+
+This may need to be revisited if we build and deliver etcd in its own RPM, but that is out of scope for this enhancement.
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+TODO
+
+#### Support Procedures
+
+##### Reading etcd Logs
+With this enhancement, etcd logs will no longer show up in the MicroShift log stream, they will be in their own systemd log stream. The user/support can get these logs with `journalctl -u microshift-etcd` or similar command used for MicroShift itself.
+
+##### Backup and Restore
```

Comment:
We have a separate ticket to work on this, so let's leave this description here but not worry about fleshing out any more of the details in this enhancement.